barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -2524,6 +2609,17 @@ std::vector<Row> AvmTraceBuilder::finalize(uint32_t min_trace_size, bool range_c
         dest.avm_sha256_sha256_compression_sel = FF(1);
         dest.avm_sha256_state = src.state[0];
     }
+    //
+    // Add Poseidon2 Gadget table
+    for (size_t i = 0; i < poseidon2_trace_size; i++) {
+        auto const& src = poseidon2_trace.at(i);
+        auto& dest = main_trace.at(i);
+        dest.avm_poseidon2_clk = FF(src.clk);
+        dest.avm_poseidon2_input = src.input[0];
+        // TODO: This will need to be enabled later
```

Comment:
Unsolved TODO?

Reply:
This TODO will only get resolved when we have this fully constrained